# Descrição do Notebook
Este notebook tem por função:

- Realizar uma análise de consistência dos dados da FEBRACE
- Realizar as seguintes análises exploratórias
    - Análise da quantidade de projetos e premiações, por ano e categoria
    - Analise da quantidade de categorias e subcategorias, por ano
    - Análise da quantidade de projetos e premiaões, por UF
    - Análise da proporção de projetos e da população, por UF
    - Análise da proporção de projetos e prêmios por categoria

# Preparação do Dataframe

Import libraries

In [1]:
#data manipulation
import pandas as pd
import numpy as np
import seaborn as sns

## data viz
import seaborn as sns
%matplotlib inline
import matplotlib
import matplotlib.pyplot as plt
import plotly
import plotly.graph_objects as go
import plotly.express as px
plotly.offline.init_notebook_mode(connected=True)

Ajuste de linhas e colunas

In [2]:
#set max number of rows and columns
pd.options.display.max_columns = None
pd.options.display.max_rows = 70

Import datasets

In [3]:
#url path
url1 = 'https://raw.githubusercontent.com/andre-carneiro/PCS5787/main/Arquivos/premios.csv'
url2 = 'https://raw.githubusercontent.com/andre-carneiro/PCS5787/main/Arquivos/projetos.csv'
url3 = 'https://raw.githubusercontent.com/andre-carneiro/PCS5787/main/Arquivos/projetos_com_premios.csv'

#import dataset
df_prem = pd.read_csv(url1)
df_proj = pd.read_csv(url2)
df_proj_prem = pd.read_csv(url3)

#drop colunas indesejada
df_proj_prem.drop(['Unnamed: 0'], axis = 1, inplace = True)
df_proj.drop(['foto', "video", "poster"], axis = 1, inplace = True)

#adição coluna aux
df_proj_prem["tem_premio"] = df_proj_prem["qtd_premios"] >0

Criação do hashing (chave única)

In [4]:
#hashing
df_prem["hash"] = pd.util.hash_pandas_object(df_prem[["codigo projeto", "ano"]],
                                                   index=False, encoding="utf8")
df_proj["hash"] = pd.util.hash_pandas_object(df_proj[["codigo", "ano"]],
                                                   index=False, encoding="utf8")
df_proj_prem["hash"] = pd.util.hash_pandas_object(df_proj_prem[["codigo", "ano"]],
                                                   index=False, encoding="utf8")

Verificação dos datasets

In [5]:
#premios
df_prem

premio  \
0        ASU Walton Sustainability Solution Initiatives   
1        ASU Walton Sustainability Solution Initiatives   
2        ASU Walton Sustainability Solution Initiatives   
3     Prêmio Marinha do Brasil -2o. Lugar em Mentali...   
4                  Intel Excellence in Computer Science   
...                                                 ...   
1459  Prêmio Destaque Unidades da Federação - Rio de...   
1460  Prêmio Destaque Unidades da Federação - Rio Gr...   
1461                                  Prêmio Space Camp   
1462                 Prêmio Intel Mulheres e Tecnologia   
1463      Prêmio Mostra Nacional de Ciência de Portugal   

                                                empresa codigo projeto  \
0     Arizona State University - Rob and Melani Walt...         HUM088   
1     Arizona State University - Rob and Melani Walt...         AGR120   
2     Arizona State University - Rob and Melani Walt...         HUM099   
3     Centro de Coordenação de Estudos da Marinha em...         ENG072   
4                  Intel Excellence in Computer Science         EXA290   
...                                                 ...            ...   
1459                                            FEBRACE         SAU120   
1460                                            FEBRACE         EXA262   
1461                       Acrux Aerospace Technologies         ENG049   
1462                                    Intel do Brasil         EXA287   
1463                              Fundação da Juventude         HUM170   

     tipo do premio  premio estudante 1  premio estudante 2  \
0           Projeto                   0                   0   
1           Projeto                   0                   0   
2           Projeto                   0                   0   
3           Projeto                   0                   0   
4           Projeto                   0                   0   
...             ...                 ...                 ...   
1459        Projeto                   0                   0   
1460        Projeto                   0                   0   
1461        Projeto                   0                   0   
1462  Participantes                   0                   1   
1463        Projeto                   0                   0   

      premio estudante 3  premio orientador  premio coorientador  \
0                      0                  0                    0   
1                      0                  0                    0   
2                      0                  0                    0   
3                      0                  0                    0   
4                      0                  0                    0   
...                  ...                ...                  ...   
1459                   0                  0                    0   
1460                   0                  0                    0   
1461                   0                  0                    0   
1462                   0                  0                    0   
1463                   0                  0                    0   

      premio escola  premio local de pesquisa   ano                  hash  
0                 0                         0  2015  16580233724752982817  
1                 0                         0  2015   1482971311073618315  
2                 0                         0  2015    663621263540065278  
3                 0                         0  2015  15675054201398720615  
4                 0                         0  2015  14987214550932579066  
...             ...                       ...   ...                   ...  
1459              0                         0  2016  10665877240586278483  
1460              0                         0  2016   6610158273954735260  
1461              0                         0  2016  17877501136951391780  
1462              0                         0  2016  17721240818074921800  
1463              0                      

In [12]:
#projetos
df_proj

titulo  \
0                                Encontro de cientistas   
1     Leite adulterado, como identificá-lo na sua casa?   
2     Epifania fronteiriça: possíveis diálogos entre...   
3      Preconceitos camuflados nas falas dos humoristas   
4     Explorando as áreas das figuras geométricas pl...   
...                                                 ...   
1724  TAAPETE: tecnologia assistiva acessível para p...   
1725  O uso da pimenta dedo-de-moça no cultivo de ho...   
1726  Macadubo: plantio sustentável de café consorci...   
1727  Produção de biomoléculas bacterianas por ferme...   
1728  Aproveitamento do calor humano na idealização ...   

                                             estudantes  \
0      Matheus Minari Matias, Mikaela Miguel dos Santos   
1     Amanda Pavão Matana, Marcos Vinicius Pandolfo,...   
2      Juliana Arevalos Bordão, Gabriela Camargo Pacher   
3     Maria Laura do Nascimento, Fernanda Gomes Coel...   
4     Barbara da Costa Amoras, Diovana de Sousa da S...   
...                                                 ...   
1724                      Álvaro da Silva Couto Vasques   
1725  Ana Cláudia Santos Piñeiro, Amanda De Lorenzi ...   
1726  THIAGO SOUZA DE OLIVEIRA, MELISSA LOPES SCHMID...   
1727  Maria Vitória de Miranda Rodrigues, Milena Car...   
1728                         Vinícius Ribeiro de Moraes   

                                           orientadores  \
0                                 Victor Seiji Fujiwara   
1          Leila Augusta Friedrich, Amanda Lucia Debuss   
2            Fabrícia Carla Viviani, Eli Gomes Castanho   
3                              Marlise Maurente Machado   
4                                   Iris Maciel Pantoja   
...                                                 ...   
1724    Leandro Silva Teixeira, Gustavo de Araujo Sabry   
1725  Ana Helena Matias Saturnino, Flávia Santos Twa...   
1726                       PAULO ROBERTO DA COSTA LEMOS   
1727  Danielle Hiromi Nakagawa, Juliana Guerra de Ol...   
1728  Marco Rogério Vieira, Michael Douglas da Silva...   

                                                 escola  UF       municipio  \
0                                     Colégio Alexandra  SP       SAO PAULO   
1                      Centro de Excelência em Educação  PR        PALOTINA   
2     Instituto Federal de Educação, Ciência e Tecno...  MS      PONTA PORA   
3                                      COLÉGIO DRUMMOND  SP          LORENA   
4                            EE RUTH DE ALMEIDA BEZERRA  AP          MACAPA   
...                                                 ...  ..             ...   
1724                         IF Baiano - Campus Valença  BA         VALENCA   
1725  Escola Estadual de Ensino Fundamental General ...  RS          OSORIO   
1726                             E.E. Ângelo Scarabucci  SP          FRANCA   
1727                          IFPR - Campus Jaguariaíva  PR     JAGUARIAIVA   
1728                     Matriz Educação - Campo Grande  RJ  RIO DE JANEIRO   

                                                 resumo  \
0     A alfabetização científica é um tema muito imp...   
1     Nos últimos tempos vem-se relatando vários cas...   
2     A presente pesquisa é fruto de um projeto inti...   
3     Diante da atual polêmica acerca do papel e dos...   
4     O processo educacional se constrói por meio de...   
...                                                 ...   
1724  O mundo moderno vive em uma constante evolução...   
1725  A agricultura familiar é uma das principais fo...   
1726  O setor agropecuário brasileiro tem nível de p...   
1727  Na última década houve um aumento na produção ...   
1728  Percebe-se o uso não consciente das fontes ene...   

                                         palavras-chave categoria  \
0          alfabetização científica, ciências, didática       HUM   
1             Leite, Adulteração , Indicadores Naturais       EXA   
2     Ditadura militar brasileira (1964 - 1985), Dit...  

In [13]:
#projetos e premios
df_proj_prem

titulo  \
0                                Encontro de cientistas   
1     Leite adulterado, como identificá-lo na sua casa?   
2     Epifania fronteiriça: possíveis diálogos entre...   
3      Preconceitos camuflados nas falas dos humoristas   
4     Explorando as áreas das figuras geométricas pl...   
...                                                 ...   
1724  TAAPETE: tecnologia assistiva acessível para p...   
1725  O uso da pimenta dedo-de-moça no cultivo de ho...   
1726  Macadubo: plantio sustentável de café consorci...   
1727  Produção de biomoléculas bacterianas por ferme...   
1728  Aproveitamento do calor humano na idealização ...   

                                             estudantes  \
0      Matheus Minari Matias, Mikaela Miguel dos Santos   
1     Amanda Pavão Matana, Marcos Vinicius Pandolfo,...   
2      Juliana Arevalos Bordão, Gabriela Camargo Pacher   
3     Maria Laura do Nascimento, Fernanda Gomes Coel...   
4     Barbara da Costa Amoras, Diovana de Sousa da S...   
...                                                 ...   
1724                      Álvaro da Silva Couto Vasques   
1725  Ana Cláudia Santos Piñeiro, Amanda De Lorenzi ...   
1726  THIAGO SOUZA DE OLIVEIRA, MELISSA LOPES SCHMID...   
1727  Maria Vitória de Miranda Rodrigues, Milena Car...   
1728                         Vinícius Ribeiro de Moraes   

                                           orientadores  \
0                                 Victor Seiji Fujiwara   
1          Leila Augusta Friedrich, Amanda Lucia Debuss   
2            Fabrícia Carla Viviani, Eli Gomes Castanho   
3                              Marlise Maurente Machado   
4                                   Iris Maciel Pantoja   
...                                                 ...   
1724    Leandro Silva Teixeira, Gustavo de Araujo Sabry   
1725  Ana Helena Matias Saturnino, Flávia Santos Twa...   
1726                       PAULO ROBERTO DA COSTA LEMOS   
1727  Danielle Hiromi Nakagawa, Juliana Guerra de Ol...   
1728  Marco Rogério Vieira, Michael Douglas da Silva...   

                                                 escola  UF       municipio  \
0                                     Colégio Alexandra  SP       SAO PAULO   
1                      Centro de Excelência em Educação  PR        PALOTINA   
2     Instituto Federal de Educação, Ciência e Tecno...  MS      PONTA PORA   
3                                      COLÉGIO DRUMMOND  SP          LORENA   
4                            EE RUTH DE ALMEIDA BEZERRA  AP          MACAPA   
...                                                 ...  ..             ...   
1724                         IF Baiano - Campus Valença  BA         VALENCA   
1725  Escola Estadual de Ensino Fundamental General ...  RS          OSORIO   
1726                             E.E. Ângelo Scarabucci  SP          FRANCA   
1727                          IFPR - Campus Jaguariaíva  PR     JAGUARIAIVA   
1728                     Matriz Educação - Campo Grande  RJ  RIO DE JANEIRO   

                                                 resumo  \
0     A alfabetização científica é um tema muito imp...   
1     Nos últimos tempos vem-se relatando vários cas...   
2     A presente pesquisa é fruto de um projeto inti...   
3     Diante da atual polêmica acerca do papel e dos...   
4     O processo educacional se constrói por meio de...   
...                                                 ...   
1724  O mundo moderno vive em uma constante evolução...   
1725  A agricultura familiar é uma das principais fo...   
1726  O setor agropecuário brasileiro tem nível de p...   
1727  Na última década houve um aumento na produção ...   
1728  Percebe-se o uso não consciente das fontes ene...   

                                         palavras-chave categoria  \
0          alfabetização científica, ciências, didática       HUM   
1             Leite, Adulteração , Indicadores Naturais       EXA   
2     Ditadura militar brasileira (1964 - 1985), Dit...  

Import informações das UFs

In [60]:
path = r'C:\Users\AndreCarneiro\Desktop\[USP] Acadêmico\Ciência dos Dados\3. Projeto\uf_pop.xlsx'
uf = pd.read_excel(path)
uf

unidade_federativa  UF       pop        regiao
0              Rondônia  RO   1815278         Norte
1                  Acre  AC    906876         Norte
2              Amazonas  AM   4269995         Norte
3               Roraima  RR    652713         Norte
4                  Pará  PA   8777124         Norte
5                 Amapá  AP    877613         Norte
6             Tocantins  TO   1607363         Norte
7              Maranhão  MA   7153262      Nordeste
8                 Piauí  PI   3289290      Nordeste
9                 Ceará  CE   9240580      Nordeste
10  Rio Grande do Norte  RN   3560903      Nordeste
11              Paraíba  PB   4059905      Nordeste
12           Pernambuco  PE   9674793      Nordeste
13              Alagoas  AL   3365351      Nordeste
14              Sergipe  SE   2338474      Nordeste
15                Bahia  BA  14985284      Nordeste
16         Minas Gerais  MG  21411923       Sudeste
17       Espírito Santo  ES   4108508       Sudeste
18       Rio de Janeiro  RJ  17463349       Sudeste
19            São Paulo  SP  46649132       Sudeste
20               Paraná  PR  11597484           Sul
21       Santa Catarina  SC   7338473           Sul
22    Rio Grande do Sul  RS  11466630           Sul
23   Mato Grosso do Sul  MS   2839188  Centro-Oeste
24          Mato Grosso  MT   3567234  Centro-Oeste
25                Goiás  GO   7206589  Centro-Oeste
26     Distrito Federal  DF   3094325  Centro-Oeste

# Análise de consistência dos dados

## Análise da tipologia

Dataframe de projetos

In [17]:
df_proj.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1729 entries, 0 to 1728
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   titulo          1729 non-null   object
 1   estudantes      1729 non-null   object
 2   orientadores    1729 non-null   object
 3   escola          1729 non-null   object
 4   UF              1729 non-null   object
 5   municipio       1729 non-null   object
 6   resumo          1729 non-null   object
 7   palavras-chave  1729 non-null   object
 8   categoria       1729 non-null   object
 9   subcategoria    1729 non-null   object
 10  codigo          1729 non-null   object
 11  ano             1729 non-null   int64 
 12  hash            1729 non-null   uint64
dtypes: int64(1), object(11), uint64(1)
memory usage: 175.7+ KB


Dataframe de prêmios

In [18]:
df_prem.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1464 entries, 0 to 1463
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   premio                    1464 non-null   object
 1   empresa                   1464 non-null   object
 2   codigo projeto            1464 non-null   object
 3   tipo do premio            1464 non-null   object
 4   premio estudante 1        1464 non-null   int64 
 5   premio estudante 2        1464 non-null   int64 
 6   premio estudante 3        1464 non-null   int64 
 7   premio orientador         1464 non-null   int64 
 8   premio coorientador       1464 non-null   int64 
 9   premio escola             1464 non-null   int64 
 10  premio local de pesquisa  1464 non-null   int64 
 11  ano                       1464 non-null   int64 
 12  hash                      1464 non-null   uint64
dtypes: int64(8), object(4), uint64(1)
memory usage: 148.8+ KB


Tipologia está correta

## Análise de valores nulos

Dataframe de projetos

In [19]:
df_proj.isin([0]).sum()

titulo            0
estudantes        0
orientadores      0
escola            0
UF                0
municipio         0
resumo            0
palavras-chave    0
categoria         0
subcategoria      0
codigo            0
ano               0
hash              0
dtype: int64

Dataframe de prêmios

In [20]:
df_prem.isin([0]).sum()

premio                         0
empresa                        0
codigo projeto                 0
tipo do premio                 0
premio estudante 1          1444
premio estudante 2          1455
premio estudante 3          1460
premio orientador           1372
premio coorientador         1463
premio escola               1464
premio local de pesquisa    1464
ano                            0
hash                           0
dtype: int64

In [61]:
df_prem.isin([0]).mean()

premio                      0.000000
empresa                     0.000000
codigo projeto              0.000000
tipo do premio              0.000000
premio estudante 1          0.986339
premio estudante 2          0.993852
premio estudante 3          0.997268
premio orientador           0.937158
premio coorientador         0.999317
premio escola               1.000000
premio local de pesquisa    1.000000
ano                         0.000000
hash                        0.000000
dtype: float64

Os valores nulos são encontrados apenas nas colunas esperadas, de valores binários. Não existem premios para escola nem local de pesquisa

## Análise de valores ausentes

Dataframe de projetos

In [21]:
df_proj.isna().sum()

titulo            0
estudantes        0
orientadores      0
escola            0
UF                0
municipio         0
resumo            0
palavras-chave    0
categoria         0
subcategoria      0
codigo            0
ano               0
hash              0
dtype: int64

Dataframe de prêmios

In [22]:
df_prem.isna().sum()

premio                      0
empresa                     0
codigo projeto              0
tipo do premio              0
premio estudante 1          0
premio estudante 2          0
premio estudante 3          0
premio orientador           0
premio coorientador         0
premio escola               0
premio local de pesquisa    0
ano                         0
hash                        0
dtype: int64

Não existem valoes ausentes no dataframe

## Análise de dados duplicados 

Dataframe de premios

In [23]:
df_prem["hash"].value_counts()

17868850594435246779    9
15025528760736909711    8
14838598545391479097    8
4097150056744540478     7
14854023866383566528    7
                       ..
1608531580940096205     1
1544690225938979304     1
655641955251275518      1
7040363324458149825     1
12348136205284388507    1
Name: hash, Length: 715, dtype: int64

Projeto com maior quantidade de prêmios

In [24]:
df_prem[df_prem["hash"] == 17868850594435246779]

premio  \
211  Society for Science & the Public Award for Com...   
248     ASU Walton Sustainability Solution Initiatives   
259  Prêmio Centro Estadual de Educação Tecnológica...   
273  Prêmio ABRIC de excelência em iniciação cientí...   
275                              Prêmio Revista ECO 21   
279                     Prêmio Nave à Vela de Inovação   
339       Prêmio Destaque Unidades da Federação - Pará   
411            1o. Lugar em Ciências Sociais Aplicadas   
422                                     Prêmio FEROCIT   

                                               empresa codigo projeto  \
211                   Society for Science & the Public         SOC110   
248  Arizona State University - Rob and Melani Walt...         SOC110   
259                                 Centro Paula Souza         SOC110   
273  Associação Brasileira de Incentivo à Ciência -...         SOC110   
275                                     Revista ECO 21         SOC110   
279                                        Nave à Vela         SOC110   
339                                            FEBRACE         SOC110   
411                                            FEBRACE         SOC110   
422  FEROCIT - Ferocit. Feira de Rondônia de Cientí...         SOC110   

    tipo do premio  premio estudante 1  premio estudante 2  \
211        Projeto                   0                   0   
248        Projeto                   0                   0   
259        Projeto                   0                   0   
273        Projeto                   0                   0   
275        Projeto                   0                   0   
279        Projeto                   0                   0   
339        Projeto                   0                   0   
411        Projeto                   0                   0   
422        Projeto                   0                   0   

     premio estudante 3  premio orientador  premio coorientador  \
211                   0                  0                    0   
248                   0                  0                    0   
259                   0                  0                    0   
273                   0                  0                    0   
275                   0                  0                    0   
279                   0                  0                    0   
339                   0                  0                    0   
411                   0                  0                    0   
422                   0                  0                    0   

     premio escola  premio local de pesquisa   ano                  hash  
211              0                         0  2018  17868850594435246779  
248              0                         0  2018  17868850594435246779  
259              0                         0  2018  17868850594435246779  
273              0                         0  2018  17868850594435246779  
275              0                         0  2018  17868850594435246779  
279              0                         0  2018  17868850594435246779  
339              0                         0  2018  17868850594435246779  
411              0                         0  2018  17868850594435246779  
422              0                         0  2018  17868850594435246779

Os valores "duplicados" obtidos pela chave única são referentes à quantidade de premios recebidos, dado que a chave única é composta pelo ano e pelo código do projeto

Dataframe de projetos

In [25]:
df_proj["hash"].value_counts()

7947670427794332256     1
3921639381186713713     1
11974572982187223789    1
17273279527221109684    1
12943100038073925237    1
                       ..
12398822307115963625    1
6067624477070450362     1
6434010423980004145     1
8569278486800323163     1
18355898103651409243    1
Name: hash, Length: 1729, dtype: int64

Não existem valores duplicados pro dataframe de projetos

## Conclusão da consistência 

Os dados estão consistentes em termos de tipologia, ausência e valores nulos e valores duplicados, e já podem ser utilizados para análise exploratória

# Análise Exploratória dos Dados

## Resumo Estatístico

### Variáveis Categóricas

In [26]:
#describe method
df_proj.describe(include = object)

titulo  \
count                                                1729   
unique                                               1723   
top     Presença de fibras têxteis sintéticas em águas...   
freq                                                    2   

                         estudantes                 orientadores  \
count                          1729                         1729   
unique                         1689                         1509   
top     João Americo Macori Barboza  Ednilson Aparecido Quarenta   
freq                              4                           10   

                                                   escola    UF  municipio  \
count                                                1729  1729       1729   
unique                                                891    27        385   
top     Fundação Escola Técnica Liberato Salzano Vieir...    SP  SAO PAULO   
freq                                                   28   462        166   

                                                   resumo  \
count                                                1729   
unique                                               1729   
top     A alfabetização científica é um tema muito imp...   
freq                                                    1   

              palavras-chave categoria    subcategoria  codigo  
count                   1729      1729            1729    1729  
unique                  1724         7              67     915  
top     Reúso, água, sistema       ENG  701 Eletrônica  ENG010  
freq                       2       443             170       6

**Conclusão:**
- Alguns projetos aparentam ter repetido ao longo dos anos (mesmo título), assim como estudantes
- Orientadores podem participar de mais de um projeto
- 891 escolas participaram, algumas com frequência maior
- São Paulo é a UF com maior participação, bem como o município de São Paulo
- Existem 7 categorias e 67 subcategorias

### Variáveis numéricas

In [27]:
#describe method
df_proj_prem[["ano","qtd_premios"]].describe(include = np.number)

ano  qtd_premios
count  1729.000000  1729.000000
mean   2017.620590     0.846732
std       1.730565     1.329505
min    2015.000000     0.000000
25%    2016.000000     0.000000
50%    2018.000000     0.000000
75%    2019.000000     1.000000
max    2020.000000     9.000000

In [28]:
#describe method
df_proj_prem["qtd_premios"].value_counts(normalize = False)

0    1014
1     346
2     165
3      98
4      61
5      30
6       9
7       3
8       2
9       1
Name: qtd_premios, dtype: int64

**Conclusão**:
- Média de premiações é de 84.7%
- Cerca de 60% dos projetos não recebem premiações, enquanto 20% recebem 1 premio e 10% recebem 2 prêmios

Quantidada de prêmios da feira

In [30]:
df_proj_prem["qtd_premios"].sum()

1464

Quantidade de projetos premiados

In [31]:
df_proj_prem["tem_premio"].sum()

715

Gráfico de distribuição de prêmios

In [32]:
fig = px.histogram(df_proj_prem, 
                   x="qtd_premios", 
#                    title = "Distribuição de prêmios por projeto",
                   template = "plotly_white",
#                   value = "count"
                  )

fig.update_layout(
    bargap=0.2,
    yaxis_title="Contagem",
    xaxis_title = "Quantidade de prêmios")

fig.update_xaxes(title_font=dict(size=25, family='Time New Roman'),
                nticks = 15,
                tickfont = dict(size = 20, family='Time New Roman'))
fig.update_yaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))

# fig.update_traces(textsrc='inside')

# fig.update_xaxes(
#         tickangle = 90,
#         title_text = "Month",
#         title_font = {"size": 20},
#         title_standoff = 25)

fig.show()

## Análise da quantidade de projetos e premiações por ano e categoria

### Projetos e premiações por ano

#### Série Temporal da quantidade de projetos, premios e projetos premiados

Criação do df

In [33]:
#df auxiliar
df_proj_ano = df_proj_prem[["ano","qtd_premios","tem_premio"]].\
              groupby("ano").\
              agg({"ano": "count", 
                   "qtd_premios": sum, 
                   "tem_premio":[sum,"mean"]})
df_proj_ano

#reset multilevel column
df_proj_ano.columns = df_proj_ano.columns.droplevel()

#rename column
df_proj_ano.columns = ["qtd_projetos", "qtd_premios", "proj_premiados" ,"media_premios"]

#reset index
df_proj_ano.reset_index(inplace = True)

#print df
df_proj_ano.rename({"qtd_projetos": "Quantidade de Projetos",
                   "qtd_premios": "Quantidade de Prêmios",
                   "proj_premiados": "Projetos Premiados"}, axis = 1, inplace = True)
df_proj_ano

ano  Quantidade de Projetos  Quantidade de Prêmios  Projetos Premiados  \
0  2015                     266                    211                 108   
1  2016                     270                    248                 122   
2  2017                     289                    260                 131   
3  2018                     278                    251                 113   
4  2019                     281                    279                 111   
5  2020                     345                    215                 130   

   media_premios  
0       0.406015  
1       0.451852  
2       0.453287  
3       0.406475  
4       0.395018  
5       0.376812

Plot série temporal

In [78]:
fig = px.line(df_proj_ano[["Quantidade de Projetos", "Quantidade de Prêmios", "Projetos Premiados","ano"]],
               x='ano', 
               y=df_proj_ano[["Quantidade de Projetos", "Quantidade de Prêmios", "Projetos Premiados","ano"]].columns,
#               title='Série temporal do comportamento dos projetos',
#              text = "value",
               template = "plotly_white",
               labels= dict(value = "Quantidade Total", 
                           ano = "Ano", 
                           variable = "Variáveis"))



fig.update_xaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))
fig.update_yaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))


fig.update_layout(
    legend=dict(
#         x=0,
#         y=1,
#         traceorder="reversed",
        title_font_family="Times New Roman",
        font=dict(
            family="Times New Roman",
            size=18,
            color="black"
        ),
#         bgcolor="LightSteelBlue",
#         bordercolor="Black",
#         borderwidth=2
    )
)


fig.update_layout(legend=dict(
    orientation="h",
    yanchor="bottom",
    y=-0.3,
    xanchor="right",
    x=1
))
#traces
# fig.update_traces(textposition="top center")


fig.show()

**Conclusão:**
- A quantidade de projetos se manteve constante até 2019, quando teve um salto
- A quantidade de prêmios teve um aumento e diminuição, estabelecendo-se em período próximo ao do início da análise
- A quantidade de projetos premiados se manteve relativamente constante
- O aumento da quantidade de projetos e diminuição da quantidade de premios diminuiu a média de projetos premiados

### Projetos e premiações por ano e categoria

#### Criação dos dataframes

df de quantidade de projetos, premio e quantidade de projetos premiados

In [35]:
df_proj_prem_cat_ano = df_proj_prem[["ano","categoria","hash","tem_premio","qtd_premios"]].\
            groupby(["ano", "categoria"], as_index = False).\
            agg({"hash": "count",
                 "tem_premio": "sum",
                 "qtd_premios": "sum"}).\
            rename({"hash": "qtd_projetos"}, 
                   axis = 1)
df_proj_prem_cat_ano

ano categoria  qtd_projetos  tem_premio  qtd_premios
0   2015       AGR            30          12           27
1   2015       BIO            40          20           41
2   2015       ENG            72          27           56
3   2015       EXA            48          22           39
4   2015       HUM            31          13           24
5   2015       SAU            28           7           13
6   2015       SOC            17           7           11
7   2016       AGR            16           7           13
8   2016       BIO            42          19           33
9   2016       ENG            73          36           83
10  2016       EXA            52          23           47
11  2016       HUM            45          18           35
12  2016       SAU            29          12           23
13  2016       SOC            13           7           14
14  2017       AGR            20          11           21
15  2017       BIO            46          24           37
16  2017       ENG            80          33           85
17  2017       EXA            59          24           46
18  2017       HUM            36          15           24
19  2017       SAU            23          12           23
20  2017       SOC            25          12           24
21  2018       AGR            21           8           16
22  2018       BIO            44          17           43
23  2018       ENG            75          34           73
24  2018       EXA            48          19           41
25  2018       HUM            39          18           32
26  2018       SAU            27           8           20
27  2018       SOC            24           9           26
28  2019       AGR            35          12           28
29  2019       BIO            43          15           47
30  2019       ENG            65          38           86
31  2019       EXA            50          19           50
32  2019       HUM            47          11           26
33  2019       SAU            24           6           22
34  2019       SOC            17          10           20
35  2020       AGR            40          17           24
36  2020       BIO            52          21           39
37  2020       ENG            78          30           50
38  2020       EXA            58          22           34
39  2020       HUM            59          19           26
40  2020       SAU            35          13           24
41  2020       SOC            23           8           18

df de quantidade de projetos por categoria

In [36]:
#qtd_projetos
df_proj_prem_cat_ano1 = pd.pivot_table(data = df_proj_prem_cat_ano,
                                      columns = "categoria",
                                      values = "qtd_projetos",
                                      index = "ano")

df_proj_prem_cat_ano_norm1 = df_proj_prem_cat_ano1.div(df_proj_prem_cat_ano1.sum(axis=1),axis=0).\
                             reset_index().rename_axis(None, axis = 1)
df_proj_prem_cat_ano_norm1

ano       AGR       BIO       ENG       EXA       HUM       SAU       SOC
0  2015  0.112782  0.150376  0.270677  0.180451  0.116541  0.105263  0.063910
1  2016  0.059259  0.155556  0.270370  0.192593  0.166667  0.107407  0.048148
2  2017  0.069204  0.159170  0.276817  0.204152  0.124567  0.079585  0.086505
3  2018  0.075540  0.158273  0.269784  0.172662  0.140288  0.097122  0.086331
4  2019  0.124555  0.153025  0.231317  0.177936  0.167260  0.085409  0.060498
5  2020  0.115942  0.150725  0.226087  0.168116  0.171014  0.101449  0.066667

df de quantidade de projetos premiados

In [37]:
#qtd_projetos
df_proj_prem_cat_ano2 = pd.pivot_table(data = df_proj_prem_cat_ano,
                                      columns = "categoria",
                                      values = "tem_premio",
                                      index = "ano")

df_proj_prem_cat_ano_norm2 = df_proj_prem_cat_ano2.div(df_proj_prem_cat_ano2.sum(axis=1),axis=0).\
                             reset_index().rename_axis(None, axis = 1)
df_proj_prem_cat_ano_norm2

ano       AGR       BIO       ENG       EXA       HUM       SAU       SOC
0  2015  0.111111  0.185185  0.250000  0.203704  0.120370  0.064815  0.064815
1  2016  0.057377  0.155738  0.295082  0.188525  0.147541  0.098361  0.057377
2  2017  0.083969  0.183206  0.251908  0.183206  0.114504  0.091603  0.091603
3  2018  0.070796  0.150442  0.300885  0.168142  0.159292  0.070796  0.079646
4  2019  0.108108  0.135135  0.342342  0.171171  0.099099  0.054054  0.090090
5  2020  0.130769  0.161538  0.230769  0.169231  0.146154  0.100000  0.061538

df da quantidade de prêmios

In [38]:
#qtd_projetos
df_proj_prem_cat_ano3 = pd.pivot_table(data = df_proj_prem_cat_ano,
                                      columns = "categoria",
                                      values = "qtd_premios",
                                      index = "ano")

df_proj_prem_cat_ano_norm3 = df_proj_prem_cat_ano3.div(df_proj_prem_cat_ano3.sum(axis=1),axis=0).\
                             reset_index().rename_axis(None, axis = 1)
df_proj_prem_cat_ano_norm3

ano       AGR       BIO       ENG       EXA       HUM       SAU       SOC
0  2015  0.127962  0.194313  0.265403  0.184834  0.113744  0.061611  0.052133
1  2016  0.052419  0.133065  0.334677  0.189516  0.141129  0.092742  0.056452
2  2017  0.080769  0.142308  0.326923  0.176923  0.092308  0.088462  0.092308
3  2018  0.063745  0.171315  0.290837  0.163347  0.127490  0.079681  0.103586
4  2019  0.100358  0.168459  0.308244  0.179211  0.093190  0.078853  0.071685
5  2020  0.111628  0.181395  0.232558  0.158140  0.120930  0.111628  0.083721

#### Quantidade de projetos por ano e por categoria

Análise absoluta

In [39]:
fig = px.line(df_proj_prem_cat_ano, 
              x='ano', 
              y="qtd_projetos",
              color = "categoria",
#               title='Quantidade de projetos por ano e por categoria',
#              text = "qtd_projetos",
             template = "plotly_white",
             labels= dict(categoria = "Categoria", 
                         qtd_projetos = "Quantidade de Projetos", 
                         ano = "Ano")
             )

#traces
fig.update_traces(textposition="top center")



fig.update_xaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))
fig.update_yaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))


fig.update_layout(
    legend=dict(
#         x=0,
#         y=1,
#         traceorder="reversed",
        title_font_family="Times New Roman",
        font=dict(
            family="Times New Roman",
            size=20,
            color="black"
        ),
#         bgcolor="LightSteelBlue",
#         bordercolor="Black",
#         borderwidth=2
    )
)

fig.show()

Análise relativa

In [40]:
fig = px.bar(df_proj_prem_cat_ano_norm1, 
             x="ano",
             y=['AGR', 'BIO', 'ENG', 'EXA', 'HUM', 'SAU', 'SOC'],
#              title="Proporção da quantidade de projetos por ano e por categoria",
#              cetegory_orders = {}
             text = "value",
             template = "plotly_white",
                            labels= dict(value = "Porcentagem", 
                           ano = "Ano", 
                           variable = "Variáveis"))
            

fig.update_traces(texttemplate='%{text:.2f}')

fig.update_xaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))
fig.update_yaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))


fig.update_layout(
    legend=dict(
#         x=0,
#         y=1,
#         traceorder="reversed",
        title_font_family="Times New Roman",
        font=dict(
            family="Times New Roman",
            size=20,
            color="black"
        ),
#         bgcolor="LightSteelBlue",
#         bordercolor="Black",
#         borderwidth=2
    )
)

fig.show()

- Maiores concentrações de projetos submetidos de engenharia, exatas, bio e humanas
- Houve aumento absoluto na quantidade de projetos de agro e de humanas, e aumento relativo no de humanas
- Engenharia diminuiu sua proporção, apesardo aumento da quantidade de projetos, o que pode indicar uma diversificação dos tipos de projetos submetidos

#### Quantidade de projetos premiados por ano e por categoria

Análise absoluta

In [41]:
fig = px.line(df_proj_prem_cat_ano, 
              x='ano', 
              y="tem_premio",
              color = "categoria",
              title='Quantidade de projetos premiados por ano e por categoria',
             text = "tem_premio")

#traces
fig.update_traces(textposition="top center")

fig.show()

Análise relativa

In [42]:
fig = px.bar(df_proj_prem_cat_ano_norm2, 
             x="ano",
             y=['AGR', 'BIO', 'ENG', 'EXA', 'HUM', 'SAU', 'SOC'],
             title="Proporção da quantidade de projetos premiados por ano e por categoria",
             text = "value"
            )

fig.update_traces(texttemplate='%{text:.2f}')

fig.show()

- Apesar das oscilações, a quantidade de projetos premiados por categoria não apresentou tendências claras de aumento ou de diminuição

#### Quantidade de premios por ano e por categoria

Análise absoluta

In [43]:
fig = px.line(df_proj_prem_cat_ano, 
              x='ano', 
              y="qtd_premios",
              color = "categoria",
              title='Quantidade de prêmios por ano e por categoria',
             text = "qtd_premios")

#traces
fig.update_traces(textposition="top center")

fig.show()

Análise relativa

In [44]:
fig = px.bar(df_proj_prem_cat_ano_norm3, 
             x="ano",
             y=['AGR', 'BIO', 'ENG', 'EXA', 'HUM', 'SAU', 'SOC'],
             title="Proporção da quantidade de prêmios por ano e por categoria",
             text = "value"
            )

fig.update_traces(texttemplate='%{text:.2f}')

fig.show()

- Como no último ano houve diminuição da quantidade de projetos premiados, ocorreu também uma queda proporcional
- Antes da diminuição, todas as categorias obtiveram aumento de quantidade de prêmios

## Exemplo de Análise por ano, categoria e subcategoria

Criação dos dataframes

In [45]:
#df auxiliar
df_proj_cat_sub = df_proj_prem[["ano","categoria", "subcategoria","qtd_premios","tem_premio"]].\
              groupby(["ano","categoria", "subcategoria"]).\
              agg({"categoria": "count",
                   "tem_premio":sum,
                   "qtd_premios": sum})

#reset multilevel column
# df_proj_cat_sub.columns = df_proj_cat_sub.columns.droplevel()

#rename column
df_proj_cat_sub.columns = ["qtd_projetos", "qtd_proj_premiados", "qtd_premios"]

#reset index
df_proj_cat_sub.reset_index(inplace = True)

#print df
df_proj_cat_sub

ano categoria                                    subcategoria  \
0    2015       AGR                                   401 Agronomia   
1    2015       AGR  402 Recursos Florestais e Engenharia Florestal   
2    2015       AGR                         403 Engenharia Agrícola   
3    2015       AGR   406 Recursos Pesqueiros e Engenharia de Pesca   
4    2015       AGR           407 Ciência e Tecnologia de Alimentos   
..    ...       ...                                             ...   
305  2020       SOC              505 Planejamento Urbano e Regional   
306  2020       SOC                                 509 Comunicação   
307  2020       SOC                              510 Serviço Social   
308  2020       SOC                          511 Economia Domestica   
309  2020       SOC                          512 Desenho Industrial   

     qtd_projetos  qtd_proj_premiados  qtd_premios  
0              10                   3            4  
1               1                   0            0  
2               6                   4           12  
3               1                   0            0  
4              12                   5           11  
..            ...                 ...          ...  
305             5                   4           11  
306             4                   1            1  
307             7                   1            3  
308             3                   1            2  
309             1                   1            1  

[310 rows x 6 columns]

Série temporal da quantidade de projetos

In [46]:
fig = px.line(df_proj_cat_sub[df_proj_cat_sub["categoria"] == "EXA"], 
              x='ano', 
              y="qtd_projetos",
              title='Série temporal da quantidade de projetos',
              color = "subcategoria",
              text = "qtd_projetos"
             )

#traces
fig.update_traces(textposition="top center")

fig.show()

Série temporal da quantidade de premios

In [47]:
fig = px.line(df_proj_cat_sub[df_proj_cat_sub["categoria"] == "EXA"], 
              x='ano', 
              y="qtd_premios",
              title='Série temporal da quantidade de premios',
              color = "subcategoria",
             text = "qtd_premios"
             )

#traces
fig.update_traces(textposition="top center")

fig.show()

Série temporal da quantidade de projetos premiados

In [48]:
fig = px.line(df_proj_cat_sub[df_proj_cat_sub["categoria"] == "EXA"], 
              x='ano', 
              y="qtd_proj_premiados",
              title='Série temporal da quantidade de projetos premiados',
              color = "subcategoria",
             text = "qtd_proj_premiados"
             )

#traces
fig.update_traces(textposition="top center")

fig.show()

Dataframe geral para todas as categorias e subcategorias

In [49]:
#df auxiliar
df_proj_cat_sub = df_proj_prem[["categoria", "subcategoria","qtd_premios","tem_premio"]].\
              groupby(["categoria", "subcategoria"]).\
              agg({"categoria": "count",
                   "tem_premio":[sum,"mean"],
                   "qtd_premios": [sum, "mean"]})

#reset multilevel column
df_proj_cat_sub.columns = df_proj_cat_sub.columns.droplevel()

#rename column
df_proj_cat_sub.columns = ["qtd_projetos", "qtd_proj_premiados", "media_proj_premiados", "qtd_premios", "media_premios_por_proj"]

#reset index
df_proj_cat_sub.reset_index(inplace = True)

#print df
df_proj_cat_sub

categoria                                    subcategoria  qtd_projetos  \
0        AGR                                   401 Agronomia            59   
1        AGR  402 Recursos Florestais e Engenharia Florestal            12   
2        AGR                         403 Engenharia Agrícola            31   
3        AGR                                   404 Zootecnia             6   
4        AGR                        405 Medicina Veterinária             4   
5        AGR   406 Recursos Pesqueiros e Engenharia de Pesca             4   
6        AGR           407 Ciência e Tecnologia de Alimentos            46   
7        BIO                              201 Biologia Geral            50   
8        BIO                                    202 Genética             4   
9        BIO                                    203 Botânica            26   
10       BIO                                    204 Zoologia            10   
11       BIO                                    205 Ecologia            69   
12       BIO                                  207 Fisiologia             5   
13       BIO                                  208 Bioquímica            35   
14       BIO                                   209 Biofísica             4   
15       BIO                                210 Farmacologia            16   
16       BIO                                  211 Imunologia             1   
17       BIO                               212 Microbiologia            43   
18       BIO                               213 Parasitologia             4   
19       ENG                                  701 Eletrônica           170   
20       ENG                               702 Eletrotécnica            52   
21       ENG                                    703 Mecânica            42   
22       ENG                                     704 Química            26   
23       ENG                                       705 Civil            37   
24       ENG                                    706 de Minas             1   
25       ENG                  707 de Materiais e Metalúrgica            37   
26       ENG                                   708 Sanitária            23   
27       ENG                                 709 de Produção            11   
28       ENG                              711 de Transportes             5   
29       ENG                            712 Naval e Oceânica             9   
30       ENG                                 713 Aerospacial             5   
31       ENG                                   714 Biomédica            25   
32       EXA                                  101 Matemática            16   
33       EXA                                 103 Estatística             1   
34       EXA                       104 Ciência da Computação           115   
35       EXA                                  105 Astronomia             6   
36       EXA                                      106 Física            35   
37       EXA                                     107 Química           133   
38       EXA                                  108 Geociência             8   
39       EXA                                     704 Química             1   
40       HUM                                   601 Filosofia             3   
41       HUM                                  602 Sociologia            46   
42       HUM                                603 Antropologia             7   
43       HUM                                    605 História            29   
44       HUM                                   606 Geografia            18   
45       HUM                                  607 Psicologia            24   
46       HUM                                    608 Educação           118   
47       HUM                            609 Ciência Política            12   
48       SAU                                    301 Medicina            31   
49       SAU                                 302 Odontologia             3   
50       SAU             

## Análise da quantidade de projetos e premiações por ano e UF

### Análise de projetos e premiações totais por UF

Criação do dataframe 

In [50]:
#df auxiliar
df_proj_uf = df_proj_prem[["UF","qtd_premios","tem_premio"]].\
              groupby("UF").\
              agg({"UF": "count",
                   "tem_premio":[sum,"mean"],
                   "qtd_premios": [sum, "mean"]})

#reset multilevel column
df_proj_uf.columns = df_proj_uf.columns.droplevel()

#rename column
df_proj_uf.columns = ["qtd_projetos", "qtd_proj_premiados", "media_proj_premiados", "qtd_premios", "media_premios_por_proj"]

#reset index
df_proj_uf.reset_index(inplace = True)

#print df
df_proj_uf

UF  qtd_projetos  qtd_proj_premiados  media_proj_premiados  qtd_premios  \
0   AC             5                   5              1.000000            7   
1   AL            32                  12              0.375000           18   
2   AM            24                  12              0.500000           26   
3   AP            22                  11              0.500000           21   
4   BA           104                  33              0.317308           72   
5   CE            66                  28              0.424242           76   
6   DF            14                   6              0.428571           10   
7   ES            12                   5              0.416667            8   
8   GO            17                   6              0.352941           17   
9   MA            33                  16              0.484848           36   
10  MG           135                  41              0.303704           85   
11  MS           115                  55              0.478261          113   
12  MT            10                   7              0.700000           14   
13  PA            37                  21              0.567568           41   
14  PB            16                   8              0.500000           13   
15  PE            46                  13              0.282609           30   
16  PI            10                   6              0.600000           15   
17  PR           125                  48              0.384000          102   
18  RJ            68                  35              0.514706           72   
19  RN           104                  52              0.500000          102   
20  RO            11                   7              0.636364           16   
21  RR             9                   6              0.666667            9   
22  RS           180                  78              0.433333          175   
23  SC            44                  19              0.431818           34   
24  SE             6                   5              0.833333            9   
25  SP           462                 170              0.367965          327   
26  TO            22                  10              0.454545           16   

    media_premios_por_proj  
0                 1.400000  
1                 0.562500  
2                 1.083333  
3                 0.954545  
4                 0.692308  
5                 1.151515  
6                 0.714286  
7                 0.666667  
8                 1.000000  
9                 1.090909  
10                0.629630  
11                0.982609  
12                1.400000  
13                1.108108  
14                0.812500  
15                0.652174  
16                1.500000  
17                0.816000  
18                1.058824  
19                0.980769  
20                1.454545  
21                1.000000  
22                0.972222  
23                0.772727  
24                1.500000  
25                0.707792  
26                0.727273

Quantidade de projetos, projetos premiados e premiações, por UF

In [51]:
fig = px.line(df_proj_uf[["qtd_projetos", "qtd_premios", "qtd_proj_premiados","UF"]], 
              x='UF', 
              y=df_proj_uf[["qtd_projetos", "qtd_premios", "qtd_proj_premiados","UF"]].columns,
              title='Quantidade de projetos, projetos premiados e premiações, por UF',
#              text = "value"
                           template = "plotly_white"
             )

#traces
fig.update_traces(textposition="top center")

fig.show()

Média de projetos premiados e prêmios por projeto, por UF

In [52]:
fig = px.line(df_proj_uf[["media_proj_premiados", "media_premios_por_proj","UF"]], 
              x='UF', 
              y=df_proj_uf[["media_proj_premiados", "media_premios_por_proj","UF"]].columns,
              title='Média de projetos premiados e prêmios por projeto, por UF',
              template = "plotly_white",
             text = "value"
             )

#traces
fig.update_traces(textposition="top center")
fig.update_traces(texttemplate='%{text:.2f}')

fig.add_hline(y=1,
              line_dash="dot",
              line_width=1,
              annotation_text="100%", 
              annotation_position="top right")

fig.show()

Entender o comportamento de algumas UFs ao longo do tempo
- Quais?
    - Mais representativas: SP, RS MG, PR, MS
    - Maiores médias de prêmios por projeto: SE, RO, PI, MT, AC
    - Maiores médias de projetos premiados: SE, MT, AC

## Análise da proporção de projetos e da população, por UF

Criação do df

In [53]:
#dataframe de UF e proporção de projetos
df_uf_aux = df_proj_prem[["UF", "municipio"]].\
              groupby("UF").\
              agg("count")\
            .merge(uf[["UF", "pop"]], 
                   on = "UF", 
                   how = "left", 
                   copy = True)\
            .set_index("UF")\
            .rename({"municipio": "% Projetos",
                    "pop": "% População"}, axis = 1)\

df_uf = df_uf_aux.div(df_uf_aux.sum(axis = 0), axis = 1).reset_index().rename_axis(None, axis = 1)

df_uf["% Projetos"] = round(df_uf["% Projetos"]*100,2)
df_uf["% População"] = round(df_uf["% População"]*100,2)

df_uf

UF  % Projetos  % População
0   AC        0.29         0.43
1   AL        1.85         1.58
2   AM        1.39         2.00
3   AP        1.27         0.41
4   BA        6.02         7.02
5   CE        3.82         4.33
6   DF        0.81         1.45
7   ES        0.69         1.93
8   GO        0.98         3.38
9   MA        1.91         3.35
10  MG        7.81        10.04
11  MS        6.65         1.33
12  MT        0.58         1.67
13  PA        2.14         4.11
14  PB        0.93         1.90
15  PE        2.66         4.54
16  PI        0.58         1.54
17  PR        7.23         5.44
18  RJ        3.93         8.19
19  RN        6.02         1.67
20  RO        0.64         0.85
21  RR        0.52         0.31
22  RS       10.41         5.38
23  SC        2.54         3.44
24  SE        0.35         1.10
25  SP       26.72        21.87
26  TO        1.27         0.75

Plot do gráfico de barras

In [54]:
fig = px.bar(df_uf, 
             x="UF", 
             y=["% Projetos","% População"],
             template = "plotly_white",
             barmode='group',
             title = "Porcentagem de Projetos e da População",
#             text = "value"
            )

fig.update_layout(
    yaxis_title="Porcentagem")

    
fig.show()

Plot do gráfico da difereça

In [80]:
df_uf ["Porc. Proj - Porc. Pop"] = df_uf["% Projetos"] - df_uf["% População"]
fig = px.bar(df_uf, x="UF", 
             y=["Porc. Proj - Porc. Pop"],
             template = "plotly_white",
             barmode='group',
#              title = "Diferença População - Projetos",
#             text = "value"
            labels = dict(variable= "Variável"))

fig.update_layout(
    yaxis_title="Porcentagem")
# fig.update_traces(texttemplate='%{text:.1f}')

fig.update_xaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))
fig.update_yaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))


fig.update_layout(
    legend=dict(
#         x=0,
#         y=1,
#         traceorder="reversed",
        title_font_family="Times New Roman",
        font=dict(
            family="Times New Roman",
            size=20,
            color="black"
        ),
#         bgcolor="LightSteelBlue",
#         bordercolor="Black",
#         borderwidth=2
    )
)

fig.update_layout(legend=dict(
    yanchor="top",
    y=0.99,
    xanchor="left",
    x=0.01
))

# fig.update_yaxes(range=[-6, 6])
    
fig.show()

## Análise de proporção de projetos e prêmios por Categoria

Criação dos dataframes

In [81]:
df_prop_cat = df_proj_prem[["categoria","hash", "qtd_premios"]].\
            groupby([ "categoria"]).\
            agg({"hash": "count",
                 "qtd_premios": "sum"}).\
            rename({"hash": "Quantidade de Projetos",
                   "qtd_premios": "Quantidade de Prêmios"}, 
                   axis = 1)

df_prop_cat_norm = df_prop_cat.div(df_prop_cat.sum(axis=0),axis=1).\
                             reset_index().rename_axis(None, axis = 1).\
                            rename({"categoria": "Categoria",
                                   "Quantidade de Projetos": "Proporção de Projetos",
                                   "Quantidade de Prêmios": "Proporção de Prêmios"}, axis = 1)

df_prop_cat_norm

Categoria  Proporção de Projetos  Proporção de Prêmios
0       AGR               0.093696              0.088115
1       BIO               0.154425              0.163934
2       ENG               0.256217              0.295765
3       EXA               0.182186              0.175546
4       HUM               0.148641              0.114071
5       SAU               0.096009              0.085383
6       SOC               0.068826              0.077186

In [87]:
fig = px.line(df_prop_cat.reset_index().rename({"categoria": "Categoria"}, axis = 1), 
              x='Categoria', 
              y=["Quantidade de Projetos", "Quantidade de Prêmios"],
#               color = "Categoria",
#               title='Quantidade de projetos por ano e por categoria',
#              text = "qtd_projetos",
             template = "plotly_white",
              labels = dict(value = "Contagem", variable = "Variáveis")

             )

#traces
fig.update_traces(textposition="top center")



fig.update_xaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))
fig.update_yaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))

fig.update_layout(
    legend=dict(
#         x=0,
#         y=1,
#         traceorder="reversed",
        title_font_family="Times New Roman",
        font=dict(
            family="Times New Roman",
            size=20,
            color="black"
        ),
#         bgcolor="LightSteelBlue",
#         bordercolor="Black",
#         borderwidth=2
    )
)

fig.update_layout(legend=dict(
    yanchor="top",
    y=0.99,
    xanchor="left",
    x=0.8
))

fig.show()

Criação do gráfico de porcentagem

In [89]:
fig = px.line(df_prop_cat_norm, 
              x='Categoria', 
              y=["Proporção de Projetos", "Proporção de Prêmios"],
#               color = "Categoria",
#               title='Quantidade de projetos por ano e por categoria',
#              text = "qtd_projetos",
             template = "plotly_white",
                            labels = dict(value = "Proporção", variable = "Variáveis")

             )

#traces
fig.update_traces(textposition="top center")

fig.update_xaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))
fig.update_yaxes(title_font=dict(size=25, family='Time New Roman'),
                tickfont = dict(size = 20, family='Time New Roman'))

fig.update_layout(
    legend=dict(
#         x=0,
#         y=1,
#         traceorder="reversed",
        title_font_family="Times New Roman",
        font=dict(
            family="Times New Roman",
            size=20,
            color="black"
        ),
#         bgcolor="LightSteelBlue",
#         bordercolor="Black",
#         borderwidth=2
    )
)

fig.update_layout(legend=dict(
    yanchor="top",
    y=0.99,
    xanchor="left",
    x=0.8
))

fig.show()